scripts/vendorSetup.js

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env node
+const { ECPair } = require('bitcoinjs-lib');
+const ChluIPFS = require('chlu-ipfs-support');
+const axios = require('axios');
+const os = require('os');
+const path = require('path');
+
+const url = process.env.URL || 'http://localhost:3000';
+const wif = process.env.WIF || null;
+
+const log = msg => console.log(msg);
+
+async function main() {
+    log('Starting Chlu IPFS');
+    const chluIpfs = new ChluIPFS({
+        type: ChluIPFS.types.vendor,
+        directory: path.join(os.tmpdir(), 'chlu-vendor-script')
+    });
+    const toWait = chluIpfs.start();
+    let keyPair;
+    if (wif) {
+        log('Opening keypair from WIF');
+        keyPair = ECPair.fromWIF(wif);
+    } else {
+        log('Generating keypair');
+        keyPair = ECPair.makeRandom();
+        log('Generated key pair. WIF = ' + keyPair.toWIF());
+    }
+    log('Waiting for ChluIPFS to be ready');
+    await toWait;
+    log('Storing Public Key to IPFS');
+    const multihash = await chluIpfs.instance.vendor.storePublicKey(keyPair.getPublicKeyBuffer());
+    log('Public Key stored at ' + multihash);
+    log('Registering vendor to ' + url); 
+    const response = await axios.post(url + '/vendors', {
+        vendorPubKeyMultihash: multihash
+    });
+    log(response.status);
+    log(response.data);
+    log('Signing Pvm key');
+    const signature = await chluIpfs.instance.vendor.signMultihash(response.data.vmPubKeyMultihash, keyPair);
+    log('Submitting signature');
+    const secondResponse = await axios.post(url + '/vendors/' + multihash + '/signature', { signature });
```

Comment:
Couple of things about the various calls to ipfs and to the url

1. Should we structure the code so that if a later call fails we have a chance to 'rollback' earlier calls?
2. What happens if the second post fails or the second save to IPFS fails? We need to put in error handling I think.
3. Should we abstract out each of these into separate methods or some other way to reduce the length of the `main` function

Reply:
1. I didn't think about a vendor being registered but not having the signature submitted. I think the best way is to clear out these vendors periodically so they don't take up space needlessly. They can always register again. Opening an issue about this https://github.com/ChluNetwork/chlu-marketplace-js/issues/5

2. I agree, this script is just for quickly setting up a vendor to use in the demo so I didn't put much thought into it. I should probably turn it into a decent CLI with nice error handling

3. Good idea :)